test/fixtures/krb5kdc-fixture/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:14.04
+FROM ubuntu:18.04
```

Comment:
This should be using the ECR one:
```
public.ecr.aws/ubuntu/ubuntu:18.04
```

Reply:
@peterzhuamazon this seems to be AWS specific, we shouldn't force people to jump off the official Docker repositories ...

Reply:
Also see: https://github.com/search?q=repo%3Aopensearch-project%2FOpenSearch+ubuntu%3A&type=code
@prudhvigodithi

Reply:
@reta We are using ECR to mitigate the Docker throttling limits at this point in time. Please let us know if you have any other ideas @dblock @CEHENKLE @nknize

Reply:
Yeahhhhh ... this is an issue, what about parametrizing the base image (build args)? Keep official by default, use AWS one for CI?

Reply:
@peterzhuamazon @kotwanikunal @prudhvigodithi Can we explore the option suggested by @reta ?

Reply:
I don't know this code well at all, but there is an example in the repository of [parameterizing this field](https://github.com/opensearch-project/OpenSearch/blob/cdfed2d1adbd255dc0576028bb0ed97d73600a49/distribution/docker/src/docker/Dockerfile#L22)

Reply:
It is not possible right now to do that in dockerfile directly, but rather with external scripts doing the docker pull call.
https://github.com/moby/moby/issues/7203
https://github.com/moby/moby/issues/11815

The only way I know of is identify how the dockerfile is being build and change like this:
```
ARG registry=docker.io
FROM ${registry}/library/ubuntu:18.04
```

Then in the call change it to `docker build --build-arg registry=<ecr registry>` which is very cumbersome.

Reply:
Using `--build-arg` is good choice but in the core code base there are multiple ways how these docker tasks are called, I see at one place it uses [docker-compose](https://github.com/search?q=repo%3Aopensearch-project%2FOpenSearch%20%20docker-compose%20&type=code). 

One other option is to use `docker save` and `docker load`, with `save` being one time operation and `docker load` can be called multiple times when the gradle docker task is executed.